src/daft-core/src/array/ops/cast.rs

```diff
@@ -623,37 +622,33 @@ fn extract_python_to_vec<
         ))),
     }?;
 
-    let py_type_fn = { PyModule::import(py, pyo3::intern!(py, "builtins"))?.getattr(pytype)? };
-    let py_memory_view = py
-        .import("builtins")?
-        .getattr(pyo3::intern!(py, "memoryview"))?;
+    let py_type_fn = { PyModule::import_bound(py, "builtins")?.getattr(pytype)? };
+    let py_memory_view = py.import_bound("builtins")?.getattr("memoryview")?;
 
     // TODO: use this to extract our the image mode
     // let py_pil_image_type = py
     //     .import("PIL.Image")
-    //     .and_then(|m| m.getattr(pyo3::intern!(py, "Image")));
+    //     .and_then(|m| m.getattr("Image"));
 
     for (i, object) in python_objects.as_arrow().iter().enumerate() {
         if let Some(object) = object {
-            let object = object.into_py(py);
-            let object = object.as_ref(py);
+            let object = object.bind(py);
 
             let supports_buffer_protocol = py_memory_view.call1((object,)).is_ok();
-            let supports_array_interface_protocol =
-                object.hasattr(pyo3::intern!(py, "__array_interface__"))?;
-            let supports_array_protocol = object.hasattr(pyo3::intern!(py, "__array__"))?;
+            let supports_array_interface_protocol = object.hasattr("__array_interface__")?;
+            let supports_array_protocol = object.hasattr("__array__")?;
 
             if supports_buffer_protocol
                 || supports_array_interface_protocol
                 || supports_array_protocol
             {
                 // Path if object supports buffer/array protocols.
-                let np_as_array_fn = py.import("numpy")?.getattr(pyo3::intern!(py, "asarray"))?;
+                let np_as_array_fn = py.import_bound("numpy")?.getattr("asarray")?;
                 let pyarray = np_as_array_fn.call1((object,))?;
                 let (num_values, shape_size) = append_values_from_numpy(
                     pyarray,
                     i,
-                    from_numpy_dtype,
+                    from_numpy_dtype.clone(),
```

Comment:
do we have to clone this or can we adapt `append_values_from_numpy` to take a bound func

Reply:
You're right, I don't think it needs a clone of the bound. I changed it to just take a `&Bound<...>`